tests/test_dia_collection.py

```diff
@@ -0,0 +1,172 @@
+
+
+def create_test_dia_objects(n_objects=5, n_src=5, scatter_arcsec=1.0,
+                            start_id=0, start_angle_degrees=0.0):
+    """ Create DIAObjects with a specified number of DIASources attached.
+
+    Parameters
+    ----------
+    n_objects : int
+        Number of DIAObjects to generate.
+    n_src : int
+        Number of DIASources to generate for each DIAObject.
+    scatter_arcsec : float
+        Scatter to add to the position of each DIASource.
+
+    Returns
+    -------
+    A list of DIAObjects
+    """
+    output_dia_objects = []
+    for obj_idx in range(n_objects):
+        src_cat = create_test_dia_sources(n_src)
+        for src_idx in range(n_src):
+            edit_source_record(
+                src_cat[src_idx], start_id + n_src * obj_idx + src_idx,
+                start_angle_degrees + 0.1 * obj_idx,
+                start_angle_degrees + 0.1 * obj_idx,
+                scatter_arcsec)
+        output_dia_objects.append(DIAObject(src_cat))
+    return output_dia_objects
+
+
+def edit_source_record(src, src_id, ra_degrees, dec_degrees, scatter_arcsec):
+    """ Edit the center coordinate and id of a source record in place.
+
+    Parameters
+    ----------
+    src : lsst.afw.table.SourceRecord
+        Input SourceRecord to edit.
+    id : int
+        Integer value to set the record id to.
+    ra_degrees : float
+        RA value to set the record coordinate to.
+    dec_degrees : float
+        DEC value to set the record coordinate to.
+    scatter_arcsec : float
+        Arcsecond scatter to add to the position of the source record coord.
+    """
+    coord = Coord(afwGeom.Angle(ra_degrees, units=afwGeom.degrees),
+                  afwGeom.Angle(dec_degrees, units=afwGeom.degrees))
+    if scatter_arcsec > 0.0:
+        coord.offset(
+            afwGeom.Angle(np.random.rand() * 360, units=afwGeom.degrees),
+            afwGeom.Angle(np.random.rand() * scatter_arcsec,
+                          units=afwGeom.arcseconds))
+    src.setCoord(coord)
+    src['id'] = src_id
+
+
+class TestDIAObjectCollection(unittest.TestCase):
+
+    def test_init(self):
+        """ Test that we can properly create a DIAObjectCollection from a list
+        of DIAObjects.
+        """
+        obj_list = create_test_dia_objects(1, 1, 0.1)
+        obj_collection = DIAObjectCollection(obj_list)
+
+        self.assertTrue(obj_collection.is_updated)
+        self.assertTrue(obj_collection.is_valid_tree)
+
+        self.assertEqual(obj_collection.get_dia_object_ids()[0], 0)
+        self.assertEqual(obj_collection.get_dia_object(0).get('id'), 0)
+
+    def test_append_and_update(self):
+        """ Test that we can add a new DIAObject to an existing
+        DIAObjectCollection.
+        """
+        obj_list = create_test_dia_objects(1, 1, 0.1)
+        obj_collection = DIAObjectCollection(obj_list)
+
+        new_dia_obj = create_test_dia_objects(1, 1, 0.1, 1, 0.1)[0]
+        obj_collection.append(new_dia_obj)
```

Comment:
If you use `extend` here, you don't have to take the first element above.

Reply:
dia_collection here is not a list object and does not have an extend method. Are you suggesting implementing such a method within DIAObjectCollection?

Reply:
Oh, I thought it was a list. Oh well!